### ISSUE TITLE
Bug: Dashboard-Folder Editors cannot create sub-folders

### ISSUE THREAD

**Issue by rgarcia89** [grafana/grafana#88120]
### What happened?

With version v11.0.0 sub-folders / nested-folders have been introduced. From my understand `Editors` on Dashboard-Folders should be able to create sub-folder within them. Same as they are able to create dashboards within these folders. However, in my environment they are not getting shown the `New folder` button when clicking on `New`.

It doesn't matter what permission I give on the dashboard-folder itself. Even with `Admin` users are not allowed to create sub-folders.

I as a grafana admin am seeing the button and can create sub-folders without any issues.

### What did you expect to happen?

They should be able to created sub-folders and therefore see the button.

### Did this work before?

Nope - was just introduced.

### How do we reproduce it?

1. Create a dashboard folder.
2. Give a user editor or admin permission on it
3. Try to create a sub-folder

### Environment (with versions)?

Grafana: 11.0.0

### Grafana platform?

Kubernetes

**Comment by rgarcia89** [comment]
It seems like the user needs to have at least `Editor` permission on the organization, which would give automatically permission to create dashboard-folders on root level as well as sub-folders in each of them.

**Comment by IevaVasiljeva** [comment]
Hi! Thanks for raising the issue! This was an oversight from our side when working on nested folders for OSS users.

We've got a fix for it [here](https://github.com/grafana/grafana/pull/91215), which will land in 11.2.0 for on-prem users, and will be rolled out to cloud users over the next few weeks. Note that the fix is currently behind a feature toggle (`accessActionSets`). If you enable the feature toggle, all of your folder editors and admins should be able to create subfolders. We're still testing this functionality, so there might be minor bugs. Once we're happy with the functionality, we will enable it by default and you will not need a feature toggle anymore (this will likely be in 11.3.0, which is currently scheduled for end of October). Until then - feel free to try out the feature toggle, and please notify us of any bugs that you run into.

**Comment by rgarcia89** [comment]
Hi @IevaVasiljeva thanks for the heads up. Sounds good to me. I will enable the feature toggle on the next version and check with my users if things work properly.

**Comment by NickAdolf** [comment]
Confirming if the feature toggle mentioned above is `only_store_access_action_sets `?

**Comment by rgarcia89** [comment]
Hi @IevaVasiljeva I have just updated to v11.2.0. Unfortunately even with feature toggle, it is not working...

![Image](https://github.com/user-attachments/assets/28ce9a9d-f88c-47f9-b492-20dff913f5b1)

I am part of the NetOps team which with `Member` permission.
NetOps has `Editor` permission on their dashboard folder.

Still I am not getting the possibility to add new folders:

![Image](https://github.com/user-attachments/assets/230d6b5e-1ff7-4033-b4a5-5b92dbb42d72)

**Comment by IevaVasiljeva** [comment]
@rgarcia89 could you try removing and readding the Editor permission (or ask an Admin to do that)? I forgot to specify that for now the fix only applies to newly added permissions. We will be adding a migration to make sure that previously added Editor and Admin permissions also allow creating subfolders, so if this is the issue it should hopefully be fixed in 11.3.0.

**Comment by NickAdolf** [comment]
> I've been following along. I have 'Viewer' and am on a team that is granted edit to a folder. With the feature toggle on I'm not able to create new subfolders unless I add `folders:create` to the role we have assigned to the team. When doing so I'm also able to add folders to the root as well, which is what we are trying to avoid.
> 
> I hand removed all permissions to the folder and replaced them with no luck.

This is from v11.1.1

**Comment by NickAdolf** [comment]
> I've been following along. I have 'Viewer' and am on a team that is granted edit to a folder. With the feature toggle on I'm not able to create new subfolders unless I add `folders:create` to the role we have assigned to the team. When doing so I'm also able to add folders to the root as well, which is what we are trying to avoid.
> 
> I hand removed all permissions to the folder and replaced them with no luck.
> 
> This is from v11.1.1

I see the reference to 11.2 above, I should have an instance of that to test with next week.

**Comment by IevaVasiljeva** [comment]
> > I've been following along. I have 'Viewer' and am on a team that is granted edit to a folder. With the feature toggle on I'm not able to create new subfolders unless I add `folders:create` to the role we have assigned to the team. When doing so I'm also able to add folders to the root as well, which is what we are trying to avoid.
> > I hand removed all permissions to the folder and replaced them with no luck.
> > This is from v11.1.1
> 
> I see the reference to 11.2 above, I should have an instance of that to test with next week.

Hey @NickAdolf! Yes, the fix is there only starting from 11.2.0.

**Comment by rgarcia89** [comment]
@IevaVasiljeva you are right. Reading them as Editor did the trick. Thanks a lot for fixing it 👍
